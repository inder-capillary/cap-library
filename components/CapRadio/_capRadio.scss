@import "~@capillarytech/cap-style-guide/styles/main";
@import "../styles/_variables.scss";

$radioButton:cap-radio-button;
.ant-radio.#{$radioButton}{
    .ant-radio-checked .ant-radio-inner {
        background-color:map-get($CAP_PRIMARY, base);
    }
    
    .ant-radio-inner:after {
        background-color:$CAP_WHITE;
    
    }
}
