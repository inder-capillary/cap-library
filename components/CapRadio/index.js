/**
* CapRadio
*/
import React from "react";
import PropTypes from 'prop-types';
import { Radio } from "antd";
import "./_capRadio.scss";
const classNames = require('classnames');
export default class CapRadio extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {className, children, ...rest} = this.props;
    return (
      <Radio {...rest} className={classNames("cap-radio-button", className)}>
        {children}
      </Radio>
    );
  }
}

CapRadio.propTypes = {
  children: PropTypes.any,
  className: PropTypes.string,
};
