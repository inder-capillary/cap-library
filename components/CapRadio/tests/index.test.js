import renderer from 'react-test-renderer';
import React from 'react'; // eslint-disable-line no-unused-vars

import CapRadio from '../index';

describe('<CapRadio />', () => {
  it('CapRadio renders correctly', () => {
    const tree = renderer.create(<CapRadio />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
