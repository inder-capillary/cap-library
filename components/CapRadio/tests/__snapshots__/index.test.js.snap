// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapRadio /> CapRadio renders correctly 1`] = `
<label
  className="cap-radio-v2 ant-radio-wrapper"
>
  <span
    className="ant-radio"
    style={Object {}}
  >
    <input
      checked={false}
      className="ant-radio-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="radio"
    />
    <span
      className="ant-radio-inner"
    />
  </span>
</label>
`;
